cylc/flow/config.py

```diff
@@ -309,6 +310,22 @@ def __init__(
         self.cfg = self.pcfg.get(sparse=False)
         self.mem_log("config.py: after get(sparse=False)")
 
+        # Get the original suite run time zone if restart:
+        cycle_point_tz = getattr(self.options, 'cycle_point_tz', None)
+        if self.cfg['cylc'].get('cycle point time zone') is None:
+            if cycle_point_tz is None:
+                # Not a restart
+                cycle_point_tz = get_local_time_zone_format()
+            # This must be set before call to init_cyclers(self.cfg):
+            self.cfg['cylc']['cycle point time zone'] = cycle_point_tz
+        elif cycle_point_tz is not None:
+            LOG.warning(
+                "Specified cycle point time zone {0} is different to the "
+                "stored cycle point time zone {1} from initial run."
```

Comment:
`, the suite will run in {1}.`

Reply:
Although actually without a change it will run in `{0}` - should it run in `{1}` instead?

At the moment, the waiting task will first trigger in `{1}` then `{0}` for all subsequent triggers - not a good idea

Reply:
It should continue with the stored value shouldn't it?

Reply:
Should there be a test for the warning message? If so how do you modify the suiterc `cycle point time zone` in the test script?

Reply:
The easiest solution is to break this functionality off into it's own method and unites it.

The amazing `caplet` pytest fixture makes testing logging easy.

```python
def test_time_zone_restart(caplog):
    config = Mock()
    # setup the config object as needed
    options = Mock()
    # setup the options object as needed
    config.options = options
   
    # set up the capture of log messages
    import logging
    from cylc.flow import CYLC_LOG
    caplog.setlevel(logging.WARNING, CYLC_LOG)

    # run the thing
    cylc.flow.config.Config.the_method_to_test(config)

    # check that a warning was raised
    assert len(caplog.reccord_tuples) == 1
```

Note we don't need to test for the exact text of the warning (we've fallen into this trap before, it makes tests very fragile and no-one wants to change log messages because they know how painful it will be to fix tests, many of which use sed/awk)